src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -1329,8 +1332,37 @@ private void Hide(object parameters)
             }
             else
             {
+                // When closing the custom node workspace, close any docked python script windows if there are no unsaved changes.
+                // Show warning message on the script editor if it has unsaved changes.
+                foreach (var node in Nodes)
```

Comment:
We may want to wrap this section into a utility function in case we want to reuse this part in the future in other workflows.

Reply:
Yeah moved it to a method in DynamoViewModel.